### DISCUSSION TITLE
Dependencia externa en Application

### DISCUSSION THREAD

**Discussion by argenis96** [jasontaylordev/CleanArchitecture#599] [Ideas]
Hola, veo que en la capa Aplicación tienes un IAplicationDBcontext con DbSet.
En teoría eso viene del paquete EF Core, y esa capa no debería hacer referencia a paquetes externos ni otros proyecto que no se Domain. 
No sé supone que las referencias a servicios y paquetes de tercero debería estar en Infraestructura?

**Comment by Leanwit** [comment]
Hello Argenins96. This topic was discussed before in these threads:
https://github.com/jasontaylordev/CleanArchitecture/discussions/483
https://github.com/jasontaylordev/CleanArchitecture/discussions/482

As you said, EF Core shouldn't be referenced in the infrastructure layer; however, the owner of this project has taken some decisions to improve readability and to simplify the architecture a cost of breaking some rules of the Clean Architecture. You can also see MediatR and AutoMapper in the application layer as well. I believe one option to avoid this is implementing repository pattern as this thread suggested https://github.com/jasontaylordev/CleanArchitecture/discussions/405
